flag.go

```diff
@@ -83,6 +83,15 @@ type RequiredFlag interface {
 	IsRequired() bool
 }
 
+// RequiredFlagErr is an interface that allows users to redefine errors on required flags
+// it allows flags with user-defined errors to be backwards compatible with the Flag interface
+type RequiredFlagErr interface {
+	Flag
+
+	IsCustom() bool
```

Comment:
❓ _(blocking question)_

This is probably related to the other blocking question -- can you help me understand the motivation behind the `IsCustom` attribute here?

Reply:
There's also probably a better way to do this and I played around with removing the field but again, helps allow both default and custom messages.